rotkehlchen/tests/unit/test_globaldb_upgrades.py

```diff
@@ -141,3 +147,33 @@ def test_upgrade_v2_v3(globaldb):
             ('BIFI', 'BIFI'),
         )
         assert cursor.fetchone()[0] == 0
+
+
+@pytest.mark.parametrize('globaldb_upgrades', [[]])
+@pytest.mark.parametrize('globaldb_version', [2])
+@pytest.mark.parametrize('target_globaldb_version', [2])
+@pytest.mark.parametrize('reload_user_assets', [False])
+def test_ongoing_upgrades(globaldb: GlobalDBHandler):
+    assert globaldb.used_backup is False
+    globaldb.add_setting_value(  # Pretend that an upgrade was started
+        name='ongoing_upgrade_from_version',
+        value=2,
+    )
+    # There are no backups, so it is supposed to raise an error
+    with pytest.raises(DBUpgradeError):
+        create_globaldb(globaldb._data_directory, 0)
+
+    # Add a backup
+    shutil.copy(
+        os.path.join(
+            os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
+            'data',
+            'v2_global.db',
+        ),
+        globaldb._data_directory / 'global_data' / f'{ts_now()}_global_db_v2.backup',  # type: ignore  # _data_directory is definitely not null  # noqa: E501
+    )
+
+    globaldb = create_globaldb(globaldb._data_directory, 0)  # Now the backup should be used
+    assert globaldb.used_backup is True
+    # Check that there is no setting left
+    assert globaldb.get_setting_value('ongoing_upgrade_from_version', -1) == -1
```

Comment:
same comment for checking the size ... or some other way ... to check that the DB is the backup and not something else